packages/scripts/config/webpack.config.js

```diff
@@ -324,45 +387,46 @@ const config = {
 		! process.env.WP_NO_EXTERNALS &&
 			new DependencyExtractionWebpackPlugin(),
 	].filter( Boolean ),
-	stats: {
-		children: false,
-	},
 };
 
-// WP_DEVTOOL global variable controls how source maps are generated.
-// See: https://webpack.js.org/configuration/devtool/#devtool.
-if ( process.env.WP_DEVTOOL ) {
-	config.devtool = process.env.WP_DEVTOOL;
-}
+if ( hasExperimentalModulesFlag ) {
+	/** @type {webpack.Configuration} */
+	const moduleConfig = {
+		...baseConfig,
 
-if ( ! isProduction ) {
-	// Set default sourcemap mode if it wasn't set by WP_DEVTOOL.
-	config.devtool = config.devtool || 'source-map';
-	config.devServer = {
-		devMiddleware: {
-			writeToDisk: true,
-		},
-		allowedHosts: 'auto',
-		host: 'localhost',
-		port: 8887,
-		proxy: {
-			'/build': {
-				pathRewrite: {
-					'^/build': '',
-				},
+		entry: getWebpackEntryPoints( 'module' ),
+
+		output: {
+			...baseConfig.output,
+			module: true,
+			chunkFormat: 'module',
+			library: {
+				...baseConfig.output.library,
+				type: 'module',
 			},
 		},
+
+		plugins: [
+			new webpack.DefinePlugin( {
+				// Inject the `SCRIPT_DEBUG` global, used for development features flagging.
+				SCRIPT_DEBUG: ! isProduction,
+			} ),
+			new RenderPathsPlugin(),
+			// The WP_BUNDLE_ANALYZER global variable enables a utility that represents
+			// bundle content as a convenient interactive zoomable treemap.
+			process.env.WP_BUNDLE_ANALYZER && new BundleAnalyzerPlugin(),
+			// MiniCSSExtractPlugin to extract the CSS thats gets imported into JavaScript.
+			new MiniCSSExtractPlugin( { filename: '[name].css' } ),
+			// React Fast Refresh.
+			hasReactFastRefresh && new ReactRefreshWebpackPlugin(),
```

Comment:
I don’t think React Fast Refresh will work out of the box for viewModule.

Reply:
I think it may work, but I'm not really sure how to test. Interactivity uses preact, which I don't think is compatible with the react refresh work. I saw this issue about refresh support for preact https://github.com/preactjs/preact/issues/2184

We probably just remove this for now for modules.

Reply:
See https://github.com/WordPress/gutenberg/pull/57461/files#r1449177817.